### PR TITLE
Add Digest-MD5 mechanism

### PR DESCRIPTION
**Pull request by laura-george**
Added Digest-MD5 mechanism for systems that do not support GSSAPI (e.g. Impala)

### CODE REVIEW

File: sasl.go [hunk 1]
```diff
@@ -173,6 +177,112 @@ func (m *CramMD5Mechanism) step(challenge []byte) ([]byte, error) {
...
+	return m.step(nil)
+}
+
+func (m *DigestMD5Mechanism) step(challenge []byte) ([]byte, error) {
+	if challenge == nil {
+		return nil, nil
+	}
+
+	// Create map of challenge
+	entries := strings.Split(string(challenge), ",")
```
Comment (beltran) [inline sasl.go]: I think this way of parsing fails if some of the values have commas, for example, for this challenge:
```
nonce="rmD6R8aMYVWH+/ih9HGBr3xNGAR6o2DUxpKlgDz6gUQ=",
realm="example.org",qop="auth,auth-int,auth-conf",cipher="rc4-40,rc
4-56,rc4,des,3des",maxbuf=65536,charset=utf-8,algorithm=md5-sess
```
where `qop` has several values. This is why pure sasl uses a [custom parsing function](https://github.com/thobbs/pure-sasl/blob/master/puresasl/mechanisms.py#L339-L394).

**Review by beltran — COMMENTED** [review]
Thank you for your contribution, would you mind adding a test as well? It should be pretty straightforward, similar to one of these: https://github.com/beltran/gosasl/blob/master/sasl_test.go#L60-L75

File: sasl.go (hunk 1 above)
Comment (laura-george) [reply sasl.go]: Ah this didn't occur to me, I am going to try it out, thanks for letting me know.

Comment (laura-george) [reply sasl.go]: And yes, I will gladly write a test :)

**Comment by laura-george** [comment]
I have a quick question, would you be willing to accept PRs that make some of the fields public so that it is possible to create new mechanisms outside of the main repo? 
I'm mainly referring to MechanismConfig https://github.com/beltran/gosasl/blob/master/sasl.go#L40

**Comment by beltran** [comment]
Unfortunately I messed up while trying to push to your branch and now I don't have permissions anymore (I think due to removing your commits when pushing), so I can't fix it, sorry about that. I've opened a new pull request with your commits and one I added here: https://github.com/beltran/gosasl/pull/22, so thank you for adding the feature.
Regarding your question, I'd be happy to make the fields public if it serves some purpose.

**Pull request closed without merging**